libpod/image/utils.go

```diff
@@ -29,23 +33,57 @@ func findImageInRepotags(search imageParts, images []*Image) (*storage.Image, er
 				continue
 			}
 			_, dName, dSuspiciousTagValueForSearch := d.suspiciousRefNameTagValuesForSearch()
-			if dName == searchName && dSuspiciousTagValueForSearch == searchSuspiciousTagValueForSearch {
-				results = append(results, image.image)
+			if dSuspiciousTagValueForSearch != searchSuspiciousTagValueForSearch {
 				continue
 			}
-			// account for registry:/somedir/image
-			if strings.HasSuffix(dName, "/"+searchName) && dSuspiciousTagValueForSearch == searchSuspiciousTagValueForSearch {
-				results = append(results, image.image)
-				continue
+			c := Candidate{
+				name:  dName,
+				image: image,
+			}
+			if dName == searchName {
```

Comment:
AFAICS this can still be ambiguous because `dName` does not include the registry.

Reply:
If I used name instead of dname as the key, would that work?

Reply:
AFAICS no (two different values in the `HasSuffix` case, but primarily, there’s no need to worry about that.

You have already written code that produces the `candidates` data, and a uniqueness condition consuming that data. Why not just process the whole loop, and have a _single_ implementation of a _single_ condition? The loop is in-memory-only, with trivial cost.